### PR TITLE
Update the CLI reference doc generation script

### PR DESCRIPTION
**Pull request by cdrage**
This updates the CLI reference doc generation script as well as the
current documentation within the help menu (odo not using correct
capitalization).

### CODE REVIEW

**Comment by cdrage** [comment]
Closes part of https://github.com/openshift/odo/issues/1565 (1/2 of it at least)

**Comment by surajnarwade** [comment]
Cluster failed to start:

```
2019/05/10 20:01:47 Resolved source https://github.com/openshift/odo to master@b6d8a903, merging: #1716 7d3665ed @cdrage
2019/05/10 20:01:47 Using namespace ci-op-8zis4cs9
2019/05/10 20:01:47 Running root, [release-inputs], src, [images], [release:latest], e2e-core
2019/05/10 20:01:47 Creating namespace ci-op-8zis4cs9
2019/05/10 20:01:47 Creating rolebinding for user cdrage in namespace ci-op-8zis4cs9
2019/05/10 20:01:47 Setting a soft TTL of 1h0m0s for the namespace
2019/05/10 20:01:47 Setting a hard TTL of 12h0m0s for the namespace
2019/05/10 20:01:47 Setting up pipeline imagestream for the test
2019/05/10 20:01:47 Created secret e2e-core-cluster-profile
2019/05/10 20:01:47 Building root
2019/05/10 20:01:47 Tagged shared images from ocp/4.0:${component}, images will be pullable from registry.svc.ci.openshift.org/ci-op-8zis4cs9/stable:${component}
2019/05/10 20:01:53 Create release image registry.svc.ci.openshift.org/ci-op-8zis4cs9/release:latest
2019/05/10 20:02:40 Build root succeeded after 53s
2019/05/10 20:02:40 Building src
2019/05/10 20:03:53 Build src succeeded after 1m13s
2019/05/10 20:03:53 Executing template e2e-core
2019/05/10 20:03:53 Creating or restarting template instance
2019/05/10 20:03:53 Waiting for template instance to be ready
2019/05/10 20:03:55 Running pod e2e-core
2019/05/10 20:03:57 Container cli in pod e2e-core completed successfully
level=warning msg="Found override for ReleaseImage. Please be warned, this is not advised"
level=info msg="Consuming \"Install Config\" from target directory"
level=info msg="Creating infrastructure resources..."
level=error
level=error msg="Error: Error applying plan:"
level=error
level=error msg="1 error occurred:"
level=error msg="\t* module.dns.aws_route53_record.api_external: 1 error occurred:"
level=error msg="\t* aws_route53_record.api_external: timeout while waiting for state to become 'INSYNC' (last state: 'PENDING', timeout: 30m0s)"
level=error
level=error
level=error
level=error
level=error
level=error msg="Terraform does not automatically rollback in the face of errors."
level=error msg="Instead, your Terraform state file has been partially updated with"
level=error msg="any resources that successfully completed. Please address the error"
level=error msg="above and apply again to incrementally change your infrastructure."
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply using Terraform"
2019/05/10 21:00:57 Container setup in pod e2e-core failed, exit code 1, reason Error
Another process exited
2019/05/10 21:01:02 Container test in pod e2e-core failed, exit code 1, reason Error
2019/05/10 21:03:33 Copied 7.75Mi of artifacts from e2e-core to /logs/artifacts/e2e-core
2019/05/10 21:03:39 Ran for 1h1m51s
error: could not run steps: step e2e-core failed: template pod "e2e-core" failed: the pod ci-op-8zis4cs9/e2e-core failed after 59m34s (failed containers: setup, test): ContainerFailed one or more containers exited

Container setup exited with code 1, reason Error
---
level=warning msg="Found override for ReleaseImage. Please be warned, this is not advised"
level=info msg="Consuming \"Install Config\" from target directory"
level=info msg="Creating infrastructure resources..."
level=error
level=error msg="Error: Error applying plan:"
level=error
level=error msg="1 error occurred:"
level=error msg="\t* module.dns.aws_route53_record.api_external: 1 error occurred:"
level=error msg="\t* aws_route53_record.api_external: timeout while waiting for state to become 'INSYNC' (last state: 'PENDING', timeout: 30m0s)"
level=error
level=error
level=error
level=error
level=error
level=error msg="Terraform does not automatically rollback in the face of errors."
level=error msg="Instead, your Terraform state file has been partially updated with"
level=error msg="any resources that successfully completed. Please address the error"
level=error msg="above and apply again to incrementally change your infrastructure."
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply using Terraform"
---
Container test exited with code 1, reason Error
---
Another process exited
---
```

/retest

File: pkg/odo/cli/cli.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ const OdoRecommendedName = "odo"
 
 var (
 	odoLong = ktemplates.LongDesc(`
-Odo (OpenShift Do) is a CLI tool for running OpenShift applications in a fast and automated matter. Odo reduces the complexity of deployment by adding iterative development without the worry of deploying your source code.
+(OpenShift Do) odo is a CLI tool for running OpenShift applications in a fast and automated matter. odo reduces the complexity of deployment by adding iterative development without the worry of deploying your source code.
```
Comment (mik-dass) [inline pkg/odo/cli/cli.go]: Maybe `it reduces the ...`. The small letter at the start of the sentence looks weird.

Comment (cdrage) [reply pkg/odo/cli/cli.go]: I know, but that's how we write `odo` now :( I'll add something else to make it look a bit better. Starting a sentence without capitalization looks odd.

**Comment by girishramnani** [comment]
Ci failure not related to this PR
```
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [component create nodejs --project ebkiaxcjbj --env key=value,key1=value1] 
[odo]  ✗  this directory already contains a component
Setting current dir to: /go/src/github.com/openshift/odo/tests/integration
```

/approve

File: pkg/odo/cli/cli.go (hunk 1 above)
Comment (mik-dass) [reply pkg/odo/cli/cli.go]: Maybe `it reduces...`?

Comment (kadel) [reply pkg/odo/cli/cli.go]: > I know, but that's how we write `odo` now :( I'll add something else to make it look a bit better. Starting a sentence without capitalization looks odd.

If you start the sentence you can use Odo ;-), or `odo` if it is as the code it looks better. 
If not in `code` it should be lowercase if not at the beginning of the sentence

**Pull request merged**